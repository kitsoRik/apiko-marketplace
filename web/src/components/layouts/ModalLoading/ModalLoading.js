import React from "react";

import Loading from "../../../assets/icons/loading.svg";

import "./ModalLoading.scss";

const ModalLoading = ({
	style,
	darken = true,
	fillPercent = 100,
	className,
	...props
}) => {
	return (
		<div
			className={`modal-loading ${className}`}
			style={style}
			darken={darken ? "true" : null}
			onClick={(e) => e.preventDefault()}
			{...props}
		>
			<img
				style={{
					maxWidth: `${fillPercent}%`,
					maxHeight: `${fillPercent}%`,
				}}
				src={Loading}
				alt="Loading"
			/>
		</div>
	);
};

export default ModalLoading;
