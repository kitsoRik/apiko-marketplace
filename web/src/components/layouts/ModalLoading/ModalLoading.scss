.modal-loading {
    height: 100%;
    width: 100%;
    position: absolute;

    display: flex;
    justify-content: center;
    align-items: center;

    background-color: transparent;

    box-sizing: border-box;

    &[darken=true] {
        background-color: #0006;
    }

    & > img {
        max-height: 100%;
        max-width: 100%;
        animation: rotating 0.4s linear infinite;
    }
}