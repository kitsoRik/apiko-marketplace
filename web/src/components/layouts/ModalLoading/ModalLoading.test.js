import React from "react";
import { create } from "react-test-renderer";
import ModalLoading from "./ModalLoading";
import { mount } from "enzyme";

describe("ModalLoading", () => {
	it("should be rendered", () => {
		const component = create(<ModalLoading />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
	it("should render without darken", () => {
		const component = create(<ModalLoading darken={false} />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
	it("should render with classname", () => {
		const component = create(
			<ModalLoading className="custom=class-name" />
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
	it("should render without different size", () => {
		let component = create(<ModalLoading fillPercent={60} />);
		let json = component.toJSON();
		expect(json).toMatchSnapshot();

		component = create(<ModalLoading fillPercent={60} />);
		json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
});
