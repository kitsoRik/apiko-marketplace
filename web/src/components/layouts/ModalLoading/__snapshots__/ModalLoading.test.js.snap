// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalLoading should be rendered 1`] = `
<div
  className="modal-loading undefined"
  darken="true"
  onClick={[Function]}
>
  <img
    alt="Loading"
    src="loading.svg"
    style={
      Object {
        "maxHeight": "100%",
        "maxWidth": "100%",
      }
    }
  />
</div>
`;

exports[`ModalLoading should render with classname 1`] = `
<div
  className="modal-loading custom=class-name"
  darken="true"
  onClick={[Function]}
>
  <img
    alt="Loading"
    src="loading.svg"
    style={
      Object {
        "maxHeight": "100%",
        "maxWidth": "100%",
      }
    }
  />
</div>
`;

exports[`ModalLoading should render without darken 1`] = `
<div
  className="modal-loading undefined"
  darken={null}
  onClick={[Function]}
>
  <img
    alt="Loading"
    src="loading.svg"
    style={
      Object {
        "maxHeight": "100%",
        "maxWidth": "100%",
      }
    }
  />
</div>
`;

exports[`ModalLoading should render without different size 1`] = `
<div
  className="modal-loading undefined"
  darken="true"
  onClick={[Function]}
>
  <img
    alt="Loading"
    src="loading.svg"
    style={
      Object {
        "maxHeight": "60%",
        "maxWidth": "60%",
      }
    }
  />
</div>
`;

exports[`ModalLoading should render without different size 2`] = `
<div
  className="modal-loading undefined"
  darken="true"
  onClick={[Function]}
>
  <img
    alt="Loading"
    src="loading.svg"
    style={
      Object {
        "maxHeight": "60%",
        "maxWidth": "60%",
      }
    }
  />
</div>
`;
